### ISSUE TITLE
IO.getEnv returns invalid UTF-8 on windows

### ISSUE THREAD

**Issue by digama0** [leanprover/lean4#3895]
Minimized from [this Zulip thread](https://leanprover.zulipchat.com/#narrow/stream/270676-lean4/topic/lake.20cache.20on.20Windows.20with.20Chinese.20locale/near/432873720). It appears that `std::getenv` on windows returns data in the current code page, so putting it directly into a lean `String` is not legal (because it assumes UTF-8 encoding). Other languages have faced the same issue:

* Nim: nim-lang/Nim#20083
* Rust: [implementation](https://github.com/rust-lang/rust/blob/9782770a810a631db7f2cf3e132354352db92c3d/library/std/src/sys/pal/windows/os.rs#L296-L303)

Both of these languages take the approach of using the widestring version of the function (which is reliably UTF-16 (with potentially unpaired surrogates)) and manually converting it to UTF-8 (or WTF-8). This is a bit sad, I thought the `A`  version of the function was supposed to be UTF-8. (Maybe it is, I can't test it.)

**Comment by Kha** [comment]
Is this https://github.com/leanprover/lean4/issues/2554?

**Comment by digama0** [comment]
Unclear. (The issue description is a bit vague.) But I don't think so; from what I can tell #2546 is about a build system issue, not runtime usage of `getEnv` and related functions.

**Comment by Kha** [comment]
No, it is not a build system issue. We set the code page of `lean` to UTF-8 but not yet for other Lean executables, so seems very relevant here
